### PR TITLE
Backend Changes for design stories

### PR DESCRIPTION
**Pull request by Tarraann**

### CODE REVIEW

File: app/db/migrations/000022_alter_stories_table.down.sql [hunk 1]
```diff
@@ -0,0 +1,5 @@
+ALTER TABLE stories
```
Comment (luciferlinx101) [inline app/db/migrations/000022_alter_stories_table.down.sql:1]: Migrations are old, update after the latest

**Review by luciferlinx101 — CHANGES_REQUESTED** [review]

File: app/controllers/story_controller.go [hunk 2]
```diff
@@ -28,6 +31,42 @@ func (controller *StoryController) CreateStory(context *gin.Context) {
 	context.JSON(http.StatusOK, gin.H{"story_id": storyID})
 }
 
+func (controller *StoryController) CreateDesignStory(context *gin.Context) {
+	file, err := context.FormFile("file")
+	if err != nil {
+		context.AbortWithStatusJSON(http.StatusInternalServerError, gin.H{"error": err.Error()})
```
Comment (czezi) [inline app/controllers/story_controller.go:37]: Add return after abort

File: app/services/story_service.go [hunk 3]
```diff
@@ -69,6 +84,126 @@ func (s *StoryService) CreateStoryForProject(requestData request.CreateStoryRequ
...
+		return 0, err
+	}
+
+	// create a story
+	createdStory, err := s.storyRepo.CreateStory(story)
+	if err != nil {
+		return 0, err
+	}
+
+	s3Url, err := s.UploadFileBytesToS3(file, fileName, projectID, int(createdStory.ID))
```
Comment (czezi) [inline app/services/story_service.go:132]: We need to have localstorage along with s3 upload, refactor out the filesystem part

File: app/workflow_executors/workflow_executor.go [hunk 4]
```diff
@@ -124,16 +166,8 @@ func (we *WorkflowExecutor) Execute(workflowConfig *WorkflowConfig, args *Workfl
 			resetDBStep.WithProject(project)
 			resetDBStep.WithExecution(execution)
 			resetDBStep.WithExecutionStep(executionStep)
-			return executor.(executors.ResetDBStepExecutor).Execute(*resetDBStep)
```
Comment (czezi) [inline app/workflow_executors/workflow_executor.go:127]: Revert these changes

File: app/workflow_executors/step_executors/steps/step_names.go [hunk 5]
```diff
@@ -12,7 +12,12 @@ var (
 	SERVER_START_STEP            StepName = "SERVER_START_STEP"
 	UPDATE_CODE_FILE_STEP        StepName = "UPDATE_CODE_FILE_STEP"
 	RESET_DB_STEP                StepName = "RESET_DB_STEP"
-	PACKAGE_INSTALL_STEP         StepName = "PACKAGE_INSTALL_STEP"
```
Comment (czezi) [inline app/workflow_executors/step_executors/steps/step_names.go:15]: revert this

**Review by czezi — CHANGES_REQUESTED** [review]

File: app/llms/claude.go [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+	if apiBaseUrl == "" {
+		apiBaseUrl = "https://api.anthropic.com/v1"
+	}
+
+	return &ClaudeClient{
+		Model:            "claude-3-5-sonnet-20240620",
+		Temperature:      0.3,
+		MaxTokens:        4000,
+		TopP:             1,
+		ApiKey:           apiKey,
```
Comment (czezi) [inline app/llms/claude.go:89]: remove default api key.

File: app/llms/claude.go [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+		NumberOfResults:  1,
+		HttpClient:       client.NewHttpClient(),
+		RetryAttempts:    3,
+		BackoffFactor:    2,
+		ResponseFormat:   &ClaudeResponseFormat{Type: "text"},
+		ApiBaseUrl:       apiBaseUrl,
+	}
+}
+
+func (c *ClaudeClient) WithApiKey(apiKey string) {
```
Comment (czezi) [inline app/llms/claude.go:101]: WithApiKey will cause issues, this should be handled properly across requests. This class can't be a singleton.

File: app/workflow_executors/workflow_config.go [hunk 8]
```diff
@@ -7,4 +7,5 @@ import (
 type WorkflowConfig struct {
 	WorkflowName string
 	StepGraph    *graph.StepGraph
+	Files        []string
```
Comment (czezi) [inline app/workflow_executors/workflow_config.go:10]: Where is this getting used?

File: app/types/request/create_worspace_request.go [hunk 9]
```diff
@@ -1,12 +1,11 @@
 package request
 
 type CreateWorkspaceRequest struct {
+	StoryHashId      string  `json:"storyHashId"`
 	WorkspaceId      string  `json:"workspaceId"`
 	RemoteURL        string  `json:"remoteURL"`
 	BackendTemplate  *string `json:"backendTemplate,omitempty"`
 	FrontendTemplate *string `json:"frontendTemplate,omitempty"`
-	GitnessUserName  string  `json:"gitnessUserName"`
```
Comment (czezi) [inline app/types/request/create_worspace_request.go:8]: Revert this?

**Review by czezi — CHANGES_REQUESTED** [review]
